File: src/types/correction.ts
/**
 * Types partagés pour la page Correction.
 *
 * Extraits de src/components/correction/correction-page.tsx (phase 1 de
 * modularisation — voir worklog T2).
 */

export interface CorrectionSession {
  id: string
  statut: string
  score: number | null
  alertes: number
  needsCorrectionCount: number
  allCorrected: boolean
  autoGradedScore: number
  autoGradedTotal: number
  etudiant: { id: string; name: string; email: string }
  epreuve: {
    id: string
    titre: string
    questions: Array<{
      id: string
      questionId: string
      bareme: number
      ordre: number
      question: {
        id: string
        type: string
        enonce: string
        propositions: string[] | null
        reponseCorrecte: string | string[] | null
        difficulte: string
        // CODE-specific fields (extracted from reponseCorrecte JSON)
        langage?: string
        codeInitial?: string
        fonctionSignature?: string
        testsPublics?: Array<{ nom: string; entree: string; sortieAttendue: string; description?: string }>
        testsPrives?: Array<{ nom: string; entree: string; sortieAttendue: string; description?: string }>
      }
    }>
  }
  reponses: Array<{
    id: string
    questionId: string
    contenu: string | null
    score: number | null
    noteIA: number | null
    justificationIA: string | null
    commentaire: string | null
  }>
  resultat: {
    id: string
    scoreFinal: number
    detailParQuestion: unknown
    dateCorrection: string | null
  } | null
}

export interface EpreuveOption {
  id: string
  titre: string
  statut: string
  dateDebut: string
  dateFin: string
}

export type GradingMode = 'par-copie' | 'par-question'

export interface RubricCriterion {
  id: string
  label: string
  points: number
}

File: mobile/shared/src/commonMain/kotlin/com/sect/mobile/shared/domain/model/Correction.kt
// SECT Mobile — Correction domain models (pure Kotlin, no serialization coupling)
package com.sect.mobile.shared.domain.model

/**
 * Session à corriger par un enseignant.
 * Domain model miroir de CorrectionSessionDto (backend domain.CorrectionSession).
 *
 * Source : GET /api/correction → { sessions: [CorrectionSessionDto] }
 */
data class CorrectionSession(
    val id: String,
    val etudiantId: String,
    val etudiant: CorrectionEtudiant?,
    val etudiantNom: String,
    val etudiantEmail: String,
    val epreuveId: String,
    val epreuveTitre: String,
    val statut: String,             // SOUMISE | CORRIGEE | RETOURNEE
    val dateFin: String?,           // ISO datetime
    val score: Double?,             // score final si déjà calculé
    val reponses: List<CorrectionReponse>,
    val alertes: Int,               // nb alertes proctoring
    val needsCorrectionCount: Int,  // réponses sans note
    val allCorrected: Boolean,
    val autoGradedScore: Double,    // QCU/QCM auto-corrigés
    val autoGradedTotal: Double,
    val resultat: CorrectionResultat?,
    val epreuve: CorrectionEpreuve?
)

data class CorrectionEtudiant(
    val id: String,
    val name: String,
    val email: String
)

data class CorrectionReponse(
    val id: String,
    val questionId: String,
    val contenu: String?,           // réponse de l'étudiant
    val score: Double?,             // note attribuée (null = non noté)
    val commentaire: String?,       // feedback enseignant
    val noteIA: Double?,            // suggestion IA
    val justificationIA: String?,   // justification de l'IA
    val statusIA: String?,          // statut correction IA (polling)
    val bareme: Double,             // points max de la question
    val ordre: Int,
    val type: String?,              // QCU | QCM | QRC | CODE
    val enonce: String?
)

data class CorrectionResultat(
    val id: String,
    val scoreFinal: Double,
    val totalPossible: Double?,
    val dateCorrection: String?
)

data class CorrectionEpreuve(
    val id: String,
    val titre: String,
    val questions: List<CorrectionQuestion>
)

data class CorrectionQuestion(
    val id: String,
    val questionId: String,
    val bareme: Double,
    val ordre: Int,
    val type: String?,
    val enonce: String?
)
